example_buffer.py

```diff
@@ -132,10 +150,69 @@ def _rsync_dir(source_dir, dest_dir):
                         stderr=rsync_log)
 
 
-def fill_and_wait(bufsize=dual_net.EXAMPLES_PER_GENERATION,
+def _determine_chunk_to_make(write_dir):
+    models = fsdb.get_models()
+    # Last model is N.  N+1 is training.  We should gather games for N+2.
+    chunk_to_make = os.path.join(write_dir, str(
+        models[-1][0] + 2) + '.tfrecord.zz')
+    while tf.gfile.Exists(chunk_to_make):
+        print("Chunk for next model ({}) already exists.  Sleeping.".format(chunk_to_make))
+        time.sleep(5 * 60)
+    print("Making chunk:", chunk_to_make)
+
+    return chunk_to_make
+
+def fill_and_wait_time(bufsize=dual_net.EXAMPLES_PER_GENERATION,
+                  write_dir=None,
+                  threads=32,
+                  start_from=dt.datetime.utcnow()):
```

Comment:
dt.datetime.utcnow() is executed at python module load time, not at python function call time - in particular, all calls will use the same default value. You'll want to execute this logic inside function body if you want the latter behavior.

Reply:
shoot, nice catch.  Done.